### PR TITLE
Ban f-strings in stubs

### PR DESCRIPTION
**Pull request by AlexWaygood**
Refs #286

### CODE REVIEW

File: pyi.py [hunk 1]
```diff
@@ -1717,6 +1717,9 @@ def _Y015_error(self, node: ast.Assign | ast.AnnAssign) -> None:
         error_message = Y015.format(old_syntax=old_syntax, new_syntax=new_syntax)
         self.error(node, error_message)
 
+    def visit_JoinedStr(self, node: ast.JoinedStr) -> None:
+        self.error(node, Y052)
```
Comment (AlexWaygood) [inline pyi.py:1721]:
```suggestion
        self.error(node, Y052)
        self.generic_visit(node)
```

**Comment by JelleZijlstra** [comment]
I'm not sure I see the need for this. Sure, there's no reason to use an f-string in a stub, but there's also no reason to use a while loop, a with statement, a walrus operator, or a ternary. Are we going to add more codes for all of those things?

**Comment by AlexWaygood** [comment]
> I'm not sure I see the need for this. Sure, there's no reason to use an f-string in a stub, but there's also no reason to use a while loop, a with statement, a walrus operator, or a ternary. Are we going to add more codes for all of those things?

Yeah. I had second thoughts shortly after filing this. It was easy to do, but probably not really necessary :)

**Comment by AlexWaygood** [comment]
Blame it on the jetlag :)

**Pull request closed without merging**